src/Core/src/Handlers/WebView/WebViewHandler.Windows.cs

```diff
@@ -279,20 +273,29 @@ void InitialCookiePreloadIfNecessary(string url)
 
 			if (cookies is not null)
 			{
-				var existingCookies = GetCookiesFromPlatformStore(url);
-				foreach (HttpCookie cookie in existingCookies)
+				var existingCookies = await GetCookiesFromPlatformStore(url);
+
+				if (existingCookies.Count == 0)
+					return;
+
+				foreach (CoreWebView2Cookie cookie in existingCookies)
 				{
+					// TODO Ideally we use cookie.ToSystemNetCookie() here, but it's not available for some reason check bac later
 					if (cookies[cookie.Name] is null)
-						myCookieJar.SetCookies(uri, cookie.ToString());
+						myCookieJar.SetCookies(uri,
+							new Cookie(cookie.Name, cookie.Value, cookie.Path, cookie.Domain)
+							{
+								Expires = DateTimeOffset.FromUnixTimeMilliseconds((long)cookie.Expires).DateTime,
+								HttpOnly = cookie.IsHttpOnly,
+								Secure = cookie.IsSecure,
+							}.ToString());
 				}
 			}
 		}
 
-		HttpCookieCollection GetCookiesFromPlatformStore(string url)
+		async Task<IReadOnlyList<CoreWebView2Cookie>> GetCookiesFromPlatformStore(string url)
 		{
-			var uri = CreateUriForCookies(url);
-			var filter = new global::Windows.Web.Http.Filters.HttpBaseProtocolFilter();
-			var platformCookies = filter.CookieManager.GetCookies(uri);
+			var platformCookies = await PlatformView.CoreWebView2.CookieManager.GetCookiesAsync(url);
 
 			return platformCookies;
```

Comment:
```suggestion
			return PlatformView.CoreWebView2.CookieManager.GetCookiesAsync(url);
```
We don't need to await in this method, do we?